### DISCUSSION TITLE
Unable to serialize

### DISCUSSION THREAD

**Discussion by n20100117** [gauravreddy08/food-vision#3] [Q&A] [answered]
Unable to serialize [2.0896919 2.1128857 2.1081853] to JSON. Unrecognized type <class 'tensorflow.python.framework.ops.EagerTensor'>.

What should I do to fix it?

**Comment by gauravreddy08** [comment]
Tensorflow `2.10` and `2.11` have some bugs with EfficientNet series. Downgrade your tensorflow version to `2.9`.

**Reply by n20100117** [reply]
Thanks
I'll try it.

**Reply by n20100117** [reply]
But when I use 2.9.3, the Building the Model section give me error
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-30-8ae2d5ec4110>](https://localhost:8080/#) in <module>
      4 # Create base model
      5 input_shape = (224, 224, 3)
----> 6 base_model = tf.keras.applications.EfficientNetB1(include_top=False)
      7 
      8 # Input and Data Augmentation

4 frames
[/usr/local/lib/python3.9/dist-packages/keras/applications/efficientnet.py](https://localhost:8080/#) in EfficientNetB1(include_top, weights, input_tensor, input_shape, pooling, classes, classifier_activation, **kwargs)
    572                    classifier_activation='softmax',
    573                    **kwargs):
--> 574   return EfficientNet(
    575       1.0,
    576       1.1,

[/usr/local/lib/python3.9/dist-packages/keras/applications/efficientnet.py](https://localhost:8080/#) in EfficientNet(width_coefficient, depth_coefficient, default_size, dropout_rate, drop_connect_rate, depth_divisor, activation, blocks_args, model_name, include_top, weights, input_tensor, input_shape, pooling, classes, classifier_activation)
    332     # original implementation.
    333     # See https://github.com/tensorflow/tensorflow/issues/49930 for more details
--> 334     x = x / tf.math.sqrt(IMAGENET_STDDEV_RGB)
    335 
    336   x = layers.ZeroPadding2D(

[/usr/local/lib/python3.9/dist-packages/tensorflow/python/util/traceback_utils.py](https://localhost:8080/#) in error_handler(*args, **kwargs)
    151     except Exception as e:
    152       filtered_tb = _process_traceback_frames(e.__traceback__)
--> 153       raise e.with_traceback(filtered_tb) from None
    154     finally:
    155       del filtered_tb

[/usr/local/lib/python3.9/dist-packages/keras/layers/core/tf_op_layer.py](https://localhost:8080/#) in handle(self, op, args, kwargs)
    105         isinstance(x, keras_tensor.KerasTensor)
    106         for x in tf.nest.flatten([args, kwargs])):
--> 107       return TFOpLambda(op)(*args, **kwargs)
    108     else:
    109       return self.NOT_SUPPORTED

[/usr/local/lib/python3.9/dist-packages/keras/utils/traceback_utils.py](https://localhost:8080/#) in error_handler(*args, **kwargs)
     65     except Exception as e:  # pylint: disable=broad-except
     66       filtered_tb = _process_traceback_frames(e.__traceback__)
---> 67       raise e.with_traceback(filtered_tb) from None
     68     finally:
     69       del filtered_tb

TypeError: Exception encountered when calling layer "tf.math.truediv" (type TFOpLambda).

`x` and `y` must have the same dtype, got tf.float16 != tf.float32.

Call arguments received by layer "tf.math.truediv" (type TFOpLambda):
  • x=tf.Tensor(shape=(None, None, None, 3), dtype=float16)
  • y=tf.Tensor(shape=(3,), dtype=float32)
  • name=None

**Reply by gauravreddy08** [reply]
try `2.9.1`

**Reply by n20100117** [reply]
OK

**Reply by n20100117** [reply]
Still the same result

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-27-8ae2d5ec4110>](https://localhost:8080/#) in <module>
      4 # Create base model
      5 input_shape = (224, 224, 3)
----> 6 base_model = tf.keras.applications.EfficientNetB1(include_top=False)
      7 
      8 # Input and Data Augmentation

4 frames
[/usr/local/lib/python3.9/dist-packages/keras/applications/efficientnet.py](https://localhost:8080/#) in EfficientNetB1(include_top, weights, input_tensor, input_shape, pooling, classes, classifier_activation, **kwargs)
    572                    classifier_activation='softmax',
    573                    **kwargs):
--> 574   return EfficientNet(
    575       1.0,
    576       1.1,

[/usr/local/lib/python3.9/dist-packages/keras/applications/efficientnet.py](https://localhost:8080/#) in EfficientNet(width_coefficient, depth_coefficient, default_size, dropout_rate, drop_connect_rate, depth_divisor, activation, blocks_args, model_name, include_top, weights, input_tensor, input_shape, pooling, classes, classifier_activation)
    332     # original implementation.
    333     # See https://github.com/tensorflow/tensorflow/issues/49930 for more details
--> 334     x = x / tf.math.sqrt(IMAGENET_STDDEV_RGB)
    335 
    336   x = layers.ZeroPadding2D(

[/usr/local/lib/python3.9/dist-packages/tensorflow/python/util/traceback_utils.py](https://localhost:8080/#) in error_handler(*args, **kwargs)
    151     except Exception as e:
    152       filtered_tb = _process_traceback_frames(e.__traceback__)
--> 153       raise e.with_traceback(filtered_tb) from None
    154     finally:
    155       del filtered_tb

[/usr/local/lib/python3.9/dist-packages/keras/layers/core/tf_op_layer.py](https://localhost:8080/#) in handle(self, op, args, kwargs)
    105         isinstance(x, keras_tensor.KerasTensor)
    106         for x in tf.nest.flatten([args, kwargs])):
--> 107       return TFOpLambda(op)(*args, **kwargs)
    108     else:
    109       return self.NOT_SUPPORTED

[/usr/local/lib/python3.9/dist-packages/keras/utils/traceback_utils.py](https://localhost:8080/#) in error_handler(*args, **kwargs)
     65     except Exception as e:  # pylint: disable=broad-except
     66       filtered_tb = _process_traceback_frames(e.__traceback__)
---> 67       raise e.with_traceback(filtered_tb) from None
     68     finally:
     69       del filtered_tb

TypeError: Exception encountered when calling layer "tf.math.truediv" (type TFOpLambda).

`x` and `y` must have the same dtype, got tf.float16 != tf.float32.

Call arguments received by layer "tf.math.truediv" (type TFOpLambda):
  • x=tf.Tensor(shape=(None, None, None, 3), dtype=float16)
  • y=tf.Tensor(shape=(3,), dtype=float32)
  • name=None
![螢幕擷取畫面 2023-03-24 180323](https://user-images.githubusercontent.com/61724453/227491468-1d2aa825-b367-45a5-ab3a-1a51465fa1f6.png)

**Reply by n20100117** [reply, marked as answer]
I have solve the problem.
I use TensorFlow 2.7.0

**Comment by inikresek** [comment]
interesting
